pandas/_libs/lib.pyx

```diff
@@ -2692,7 +2692,7 @@ def maybe_convert_objects(ndarray[object] objects,
         if using_pyarrow_string_dtype() and is_string_array(objects, skipna=True):
             from pandas.core.arrays.string_ import StringDtype
 
-            dtype = StringDtype(storage="pyarrow_numpy")
+            dtype = StringDtype()
```

Comment:
There are a bunch more of such changes needed to update the code paths behind `if using_pyarrow_string_dtype()` (which is essentially `future.infer_string = True`) to properly handle the case without pyarrow installed. But given that this is not needed to actually get the tests passing (we are not testing with infer_strings enabled), would prefer keeping this for a follow-up PR.

Reply:
Doesn't this need to be `StringDtype(na_value=np.nan)` to remain equiavlent with what's in main?

Reply:
Yes, this PR stemmed from the time that the PDEP was still proposing to make NaN the default in the `StringDtype()` constructor. 
I have to update the constructors here, but decided to first add the `na_value` keyword for what already exists in main (i.e. https://github.com/pandas-dev/pandas/pull/59330)